corgi/tasks/pnc.py

```diff
@@ -123,3 +124,62 @@ def slow_fetch_pnc_sbom(purl: str, product_data, build_data, sbom_data) -> None:
 
     # Save component taxonomy
     root_component.save_component_taxonomy()
+
+
+@app.task(base=Singleton, autoretry_for=RETRYABLE_ERRORS, retry_kwargs=RETRY_KWARGS)
+def slow_handle_pnc_errata_released(erratum_id: int, erratum_status: str) -> None:
+    # Check that the erratum is released
+    if erratum_status != "SHIPPED_LIVE":
+        raise ValueError(f"Invalid status {erratum_status} for erratum {erratum_id}")
+
+    # Get the purl(s) of the related component(s) from the erratum's Notes field
+    et = ErrataTool()
+    notes = et.get_erratum_notes(erratum_id)
+
+    related_purls = set()
+    for ref in notes.get("manifest", {}).get("refs", {}):
+        if ref["type"] == "purl":
+            related_purls.add(ref["uri"])
+
+    if not related_purls:
+        raise ValueError(f"Erratum {erratum_id} had no associated purls")
+
+    # Check that there's a component matching the purls
+    root_components: set[Component] = set()
+    for purl in related_purls:
+        # There should only be one root component of SOURCE type
+        components = Component.objects.filter(meta_attr__purl_declared=purl).filter(
+            type=Component.Type.MAVEN
+        )
+        if components.count() == 0:
+            logger.warning(
+                f"Erratum {erratum_id} refers to purl {purl} which matches no components"
+            )
+            continue
+        if components.count() > 1:
+            logger.warning(
+                f"Erratum {erratum_id} refers to purl {purl} which matches {components.count()}"
```

Comment:
I think this will make three DB calls, so you could store the result of .count() to speed this up just a little bit:
```suggestion
        components_count = components.count()
        if components_count == 0:
            logger.warning(
                f"Erratum {erratum_id} refers to purl {purl} which matches no components"
            )
            continue
        elif components_count > 1:
            logger.warning(
                f"Erratum {erratum_id} refers to purl {purl} which matches {components_count}"
```
But I'm also not sure it's necessary. We have a unique constraint on the regular purl field. Even though you're using the meta_attr__purl_declared field, I'd be very surprised if we had two Maven components with different regular purls, but the same purl_declared.

One option might be to fix up our purl-generating logic for Maven components, so that we can raise an exception (during ingestion) if Corgi's purl isn't the same as SBOMer's purl_declared. Then the purl_declared will definitely be unique, and this check won't be needed.

Reply:
I have CORGI-789 planned for this sprint, and I plan to address potential discrepancies between the SBOMer purl and Corgi's generated purls as part of that. When I make that change, I 'll also remove this check.